### ISSUE TITLE
Enable --project (tsconfig.json) to be set in tslint.json

### ISSUE THREAD

**Issue by treythomas123** [palantir/tslint#3865]
Currently the path to `tsconfig.json` can only be set via the `-p`/`--project` CLI arg, so I have to run tslint like this (usually via a script):
```
tslint -p tsconfig.json
```

It'd be nice to be able to set this via `tslint.json`, so it doesn't always need to be set via CLI:
```
{
  "project": "./tsconfig.json",
  "rulesDirectory": ...
  "rules": ...
}
```

This would also be useful for editor plugins, since it removes the need for the plugin to provide the project-specific path to `tsconfig.json` when running tslint.

Related to https://github.com/palantir/tslint/issues/2788

**Comment by adidahiya** [comment]
+1, but it should be nested inside the `linterOptions` object

```
{
  "extends": "tslint:recommended",
  "linterOptions": {
    "project": "src/tsconfig.json"
  },
  "rules": {}
}
```

**Comment by slikts** [comment]
I keep getting this warning in vscode because of the lack of this feature: `Warning: The 'no-boolean-literal-compare' rule requires type information.`

**Comment by JoshuaKGoldberg** [comment]
### 💀 _**It's time!**_ 💀

**TSLint is being deprecated** and no longer accepting pull requests for major new changes or features. See #4534. 😱

If you'd like to see this change implemented, you have two choices:
* **Recommended:** Check if this is available in [ESLint](https://eslint.org) + [typescript-eslint](https://typescript-eslint.io) ✅
* _Not Recommended: Fork TSLint locally_ 🤷‍♂️

👋 It was a pleasure open sourcing with you!

_If you believe this message was posted here in error, please comment so we can re-open the issue!_

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
